src/components/ContactCard.tsx

```diff
@@ -0,0 +1,20 @@
+import { Contact } from '../types';
+
+interface ContactCardProps {
+    contact: Contact;
+    onEdit: (id: string) => void
+}
+
+export default function ContactCard({ contact, onEdit }: ContactCardProps) {
+    return (
+        <div className="card flex flex-col w-96 bg-base-100 card-md shadow-sm" key={ contact.id }>
+            <div className="card-body">
+                <h1 id="cardName" className="card-title font-bold">{ contact.name }</h1>
```

Comment:
You add the HTML `id` attribute to the `h1` and `div` elements in the `ContactCard` component.
Since you are rendering this component once for each contact, as soon as you have more than 1 contact, you will end up with duplicated HTML ids! This is invalid HTML markup.

Do you really need the `id` attribute here? Consider simply removing it.
Alternatively, if you want to keep the id, consider generating it using React's [useId](https://react.dev/reference/react/useId) hook, so each ContactCard will use unique ids.